File: sentimental-cash/cash.py
# TODO
from cs50 import get_float


def main():
    change = int(get_float("Change owed: ") * 100)
    coins = 0
    # if proper change
    if change > 0:

        while change > 0:
            # for quarters
            if change > 24:
                change -= 25
                coins += 1
                # print("quarters " + str(coins))
            # for dimes
            elif change > 9 and change < 25:
                change -= 10
                coins += 1
                # print("dimes " + str(coins))
            # for nickels
            elif change > 4 and change < 10:
                change -= 5
                coins += 1
                # print("nickels " + str(coins))
            # for pennies
            elif change > 0 and change < 5:
                change -= 1
                coins += 1
                # print("pennies " + str(coins))
        print(coins)
    # not proper change
    else:
        main()


main()

File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

int get_cents(void)
{
    int n;  //declare n
    // ask for change
    do
    {
        n = get_int("Change owed: ");
    }
    while (n < 0);  // keep asking if change is negative
    return n;
}

int calculate_quarters(int cents)
{
    int x = 0;
    // for cents above 25
    while (cents >= 25)
    {
        cents -= 25;
        x += 1;
    }
    return x;
}

int calculate_dimes(int cents)
{
    int y = 0;
    // for cents above 10 and less than 25
    while (cents >= 10)
    {
        cents -= 10;
        y += 1;
    }
    return y;
}

int calculate_nickels(int cents)
{
    int z = 0;
    // for change between 5 and 10
    while (cents >= 5)
    {
        cents -= 5;
        z += 1;
    }
    return z;
}

int calculate_pennies(int cents)
{
    // simple return for change less than 5
    return cents;
}
